packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -330,29 +333,31 @@ export function cloudflare(pluginConfig: PluginConfig = {}): vite.Plugin[] {
 				return getWorkerConfig(environment.name) !== undefined;
 			},
 			async resolveId(source, importer) {
-				if (!source.endsWith(".wasm")) {
+				const additionalModuleType = matchAdditionalModule(source);
+
+				if (!additionalModuleType) {
 					return;
 				}
 
-				const resolved = await this.resolve(source, importer);
+				const resolved = await this.resolve(cleanUrl(source), importer);
```

Comment:
I found when doing the Node.js compat stuff that manually cleaning a query off the end of an Id was a code smell that I was not really doing the right thing with how Vite expected things to work.

Can you confirm why this is needed?

Reply:
Cleaning the query params isn't always a code smell. This `cleanUrl` function is actually copied from Vite and used a lot in their internal plugins. In this case, it is because the default rules include both `.wasm` and `.wasm?module`. Therefore, we want the match to include the query param but remove it before resolving the ID. I can add a comment to clarify this.

On a related note, I'm not 100% if we need to support `.wasm?module` going forward but included it so we don't deviate from Wrangler's behaviour.

Reply:
[cf4c860](https://github.com/cloudflare/workers-sdk/pull/8387/commits/cf4c860c2b85b35f7fb469a0eaedef2c7c1a1645)